File: Leetcode/string/67_AddBinary/Solution.java
class Solution {
    public String addBinary(String a, String b) {
        int aLen = a.length(), bLen = b.length();
        if(bLen==0)
            return a;
        if(aLen==0)
            return b;
        if(a.charAt(aLen-1)=='1' && b.charAt(bLen-1)=='1'){
            return addBinary(addBinary(a.substring(0, aLen-1), b.substring(0, bLen-1)), "1")+"0";
        }else if(a.charAt(aLen-1)=='0' && b.charAt(bLen-1)=='0'){
            return addBinary(a.substring(0, aLen-1), b.substring(0, bLen-1))+"0";
        }else{
            return addBinary(a.substring(0, aLen-1), b.substring(0, bLen-1))+"1";
        }
    }
}

File: Leetcode/string/67_AddBinary/Solution.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        if not b:
            return a
        if not a:
            return b
        if a[-1]=='1' and b[-1]=='1':
            return self.addBinary(self.addBinary(a[0:-1], b[0:-1]), '1')+'0'
        elif a[-1]=='0' and b[-1]=='0':
            return self.addBinary(a[0:-1], b[0:-1])+'0'
        else:
            return self.addBinary(a[0:-1], b[0:-1])+'1'
